### PR TITLE
LITE-28705: Migrate component templates from Pug to plain HTML

### PR DESCRIPTION
**Pull request by arnaugiralt**
All components' templates have been moved to html as discussed.

Also:
- Added more strict ESLint rules for vue
- Updated dependencies, which required major work on the migration of Storybook.

### CODE REVIEW

**Review by paulinaczybir — DISMISSED** [review]

File: src/stories/Card.stories.mdx
Comment (paulinaczybir) [inline src/stories/Card.stories.mdx:1]: is this documentation still working? in this storybook docs it says that writing stories directly in MDX was deprecated: https://storybook.js.org/docs/vue/writing-docs/mdx

File: .storybook/main.js [hunk 1]
```diff
@@ -1,23 +1,30 @@
...
+
   staticDirs: ["../public"],
+
   addons: [
     "@storybook/addon-links",
     "@storybook/addon-essentials",
     "@storybook/addon-interactions",
-	"storybook-addon-designs"
+    "@storybook/addon-designs",
+    "@storybook/addon-mdx-gfm",
```
Comment (paulinaczybir) [inline .storybook/main.js:16]: From my understanding this package can be removed after the migration is done: https://storybook.js.org/addons/@storybook/addon-mdx-gfm

File: package.json [hunk 2]
```diff
@@ -5,54 +5,57 @@
...
-    "@vue/cli-plugin-babel": "^5.0.4",
-    "@vue/vue3-jest": "^28.0.0",
+    "@babel/core": "^7.23.0",
+    "@babel/eslint-parser": "^7.22.15",
+    "@storybook/addon-actions": "^7.4.5",
+    "@storybook/addon-designs": "^7.0.5",
+    "@storybook/addon-essentials": "^7.4.5",
+    "@storybook/addon-interactions": "^7.4.5",
+    "@storybook/addon-links": "^7.4.5",
+    "@storybook/addon-mdx-gfm": "^7.4.5",
```
Comment (paulinaczybir) [inline package.json:25]: From my understanding "@storybook/addon-mdx-gfm" can be removed after the migration is done: https://storybook.js.org/addons/@storybook/addon-mdx-gfm

File: .storybook/main.js [hunk 3]
```diff
@@ -1,23 +1,30 @@
 const path = require('path');
 
 module.exports = {
-  stories: 
-  ["../src/**/*.stories.@(js|jsx|ts|tsx)",
-  "../src/**/*.stories.mdx",],
+  stories: [
+    "../src/**/*.stories.@(js|jsx|ts|tsx)",
+    "../src/**/*.stories.mdx",
```
Comment (paulinaczybir) [inline .storybook/main.js:6]: to make it work with the upgraded storybook this should be changed from "../src/**/*.stories.mdx" => "../src/**/*.mdx"

Comment (arnaugiralt) [reply .storybook/main.js]: Done, thx

File: src/stories/Card.stories.mdx
Comment (arnaugiralt) [reply src/stories/Card.stories.mdx:1]: Done, thx

**Pull request merged**